src/bun.js/api/bun/socket.zig

```diff
@@ -1868,9 +1942,21 @@ fn NewSocket(comptime ssl: bool) type {
 
             const globalObject = handlers.globalObject;
             const this_value = this.getThisValue(globalObject);
+            var js_error: JSValue = .undefined;
+            if (err != 0) {
+                if (bun.C.SystemErrno.init(err)) |str| {
```

Comment:
Is there a helper you can use for this? I think there are more edgecases here and it shouldn't live in this function.

Reply:
fixed